packages/manager/apps/pci-ai-notebooks/src/pages/notebooks/create/_components/OrderFunnel.component.tsx

```diff
@@ -0,0 +1,673 @@
+
+interface OrderFunnelProps {
+  regions: ai.capabilities.Region[];
+  catalog: order.publicOrder.Catalog;
+  frameworks: ai.capabilities.notebook.Framework[];
+  editors: ai.capabilities.notebook.Editor[];
+  sshKeys: SshKey[];
+  suggestions: Suggestions[];
+}
+
+const OrderFunnel = ({
+  regions,
+  catalog,
+  frameworks,
+  editors,
+  sshKeys,
+  suggestions,
+}: OrderFunnelProps) => {
+  const model = useOrderFunnel(
+    regions,
+    catalog,
+    frameworks,
+    editors,
+    suggestions,
+  );
+  const { t } = useTranslation('pci-ai-notebooks/notebooks/create');
+  const [showAdvancedConfiguration, setShowAdvancedConfiguration] = useState(
+    false,
+  );
+  const accordionContentRef = useRef(null);
+  const cliEquivalentModale = useModale('cli');
+  const navigate = useNavigate();
+
+  const { toast } = useToast();
+  const [command, setCommand] = useState<ai.Command>({});
+
+  const addSshKeyModale = useModale('addSshKey');
+  const { addNotebook, isPending: isPendingAddNotebook } = useAddNotebook({
+    onError: (err) => {
+      toast({
+        title: t('errorCreatingNotebook'),
+        variant: 'destructive',
+        description: getAIApiErrorMessage(err),
+      });
+    },
+    onSuccess: (notebook) => {
+      toast({
+        title: t('successCreatingNotebook'),
+      });
+      navigate(`../${notebook.id}`);
+    },
+  });
+
+  const { getCommand, isPending: isPendingCommand } = useGetCommand({
+    onError: (err) => {
+      toast({
+        title: t('errorGetCommandCli'),
+        variant: 'destructive',
+        description: getAIApiErrorMessage(err),
+      });
+    },
+    onSuccess: (cliCommand) => {
+      setCommand(cliCommand);
+    },
+  });
+
+  const getCliCommand = () => {
+    const notebookInfos: ai.notebook.NotebookSpec = {
+      env: {
+        frameworkId: model.result.framework.id,
+        frameworkVersion: model.form.getValues('frameworkWithVersion.version'),
+        editorId: model.result.editor.id,
+      },
+      region: model.result.region.id,
+      unsecureHttp: model.result.unsecureHttp,
+      sshPublicKeys: model.result.sshKey,
+      labels: model.result.labels,
+    };
+
+    if (model.result.flavor.type === ai.capabilities.FlavorTypeEnum.cpu) {
+      notebookInfos.resources = {
+        flavor: model.result.flavor.id,
+        cpu: model.result.resourcesQuantity,
+      };
+    } else {
+      notebookInfos.resources = {
+        flavor: model.result.flavor.id,
+        gpu: model.result.resourcesQuantity,
+      };
+    }
+
+    if (model.result.volumes.length > 0) {
+      notebookInfos.volumes = model.result.volumes.map(
+        (volume: OrderVolumes) => ({
+          cache: volume.cache,
+          dataStore: {
+            alias: volume.dataStore.alias,
+            container: volume.dataStore.container,
+          },
+          mountPath: volume.mountPath,
+          permission: volume.permission,
+        }),
+      );
+    }
```

Comment:
This part looks like it is duplicated for both the creation and the cli equivalent. Can we move it to a helper function instead ?